mlrun/api/utils/clients/iguazio.py

```diff
@@ -89,6 +90,7 @@ def __init__(self, *args, **kwargs) -> None:
             [[1, 10], [5, None]]
         )
         self._wait_for_project_terminal_state_retry_interval = 5
+        self._igz_clients = {}
```

Comment:
why is that? because it is a singleton?

Reply:
this is mainly "hahana lemazgan" to maintain multiple igz clients for different environments **and** for caching and iguazio client so we won't need to initialize it on every request.